src/domain/service/fileUploader.service.ts

```diff
@@ -0,0 +1,238 @@
+import type { FileData } from '@domain/entities/file.js';
+import { FileTypes } from '@domain/entities/file.js';
+import type { NoteInternalId } from '@domain/entities/note.js';
+import type User from '@domain/entities/user.js';
+import { createFileId } from '@infrastructure/utils/id';
+import type FileRepository from '@repository/file.repository.js';
+import type ObjectRepository from '@repository/object.repository.js';
+import type NoteSettingsSharedMethods from './shared/noteSettings.js';
+import { MemberRole } from '@domain/entities/team';
+import { DomainError } from '@domain/entities/DomainError.js';
+
+/**
+ * File data for upload
+ */
+interface FileToUpload {
+  /**
+   * File data
+   */
+  data: FileData;
+  /**
+   * File name
+   */
+  name: string;
+  /**
+   * Mimetype of the file
+   */
+  mimetype: string;
+  /**
+   * File type
+   */
+  type: FileTypes;
+}
+
+/**
+ * File upload details, contains user id, who uploaded the file and note id, in case if file is a part of note
+ */
+interface FileUploadDetails {
+  /**
+   * User who uploaded the file
+   */
+  userId?: User['id'];
+
+  /**
+   * In case if file is a part of note, note id to identify permissions to access
+   */
+  noteId?: NoteInternalId;
+}
+
+/**
+ * File uploader service
+ */
+export default class FileUploaderService {
+  /**
+   * Repository for objects
+   */
+  private readonly objectRepository: ObjectRepository;
+
+  /**
+   * Repository for files data
+   */
+  private readonly fileRepository: FileRepository;
+
+  /**
+   * Note settings shared methods
+   */
+  private readonly noteSettingsSharedMethods: NoteSettingsSharedMethods;
+
+  /**
+   * File uploader service constructor
+   *
+   * @param objectRepository - repository for objects
+   * @param fileRepository - repository for files data
+   * @param noteSettingsSharedMethods - note settings shared methods
+   */
+  constructor(objectRepository: ObjectRepository, fileRepository: FileRepository, noteSettingsSharedMethods: NoteSettingsSharedMethods) {
+    this.noteSettingsSharedMethods = noteSettingsSharedMethods;
+    this.objectRepository = objectRepository;
+    this.fileRepository = fileRepository;
+  }
+
+  /**
+   * Upload file
+   *
+   * @param fileData - file data to upload (e.g. buffer, name, mimetype)
+   * @param details - file upload details (e.g. user id, note id)
+   */
+  public async uploadFile(fileData: FileToUpload, details?: FileUploadDetails): Promise<string> {
+    const key = createFileId();
+
+    let hasPermissions: boolean;
+
+    /**
+     * If file is a part of note, check if user has permissions to access file
+     */
+    if (details?.noteId !== undefined) {
+      hasPermissions = await this.hasPermissionsToAccessFileInNote(details.noteId, MemberRole.Write, details.userId);
+    } else {
+      hasPermissions = true;
+    }
```

Comment:
I think we can use a policy for that

Reply:
removed checking permissions from service